shopelectro/views/catalog.py

```diff
@@ -77,6 +84,26 @@ def get_context_data(self, **kwargs):
             'group_tags_pairs': group_tags_pairs
         }
 
+    def render_siblings_on_404(
+        self, request, **url_kwargs
+    ) -> typing.Union[http.Http404, None]:
+        """Try to render removed product's siblings on it's 404 page."""
+        product_inactive_qs = models.Product.objects.filter(
+            **{self.slug_field: url_kwargs['product_vendor_code']},
+            category__isnull=False,
+            page__is_active=False
+        )
+        if product_inactive_qs:
+            product = product_inactive_qs.first()
+            related_products = models.Product.objects.filter(
+                category=product.category,
+                page__is_active=True
+            )[10:]
```

Comment:
>related_products = models.Product.objects.filter(..)**[10:]**

maybe, did you mean here first ten products instead?
`related_products = models.Product.objects.filter(..)[:10]`

Reply:
Done.